app/scripts/shutterbug-worker.js

```diff
@@ -99,10 +99,42 @@ ShutterbugWorker.prototype.getHtmlFragment = function(callback) {
     });
 
     if (element.is('canvas')) {
-      element = replacementImgs[0];
+      element = replacementCanvasImgs[0];
     } else {
       element.find('canvas').each(function(i, elem) {
-        $(elem).replaceWith(replacementImgs[i]);
+        $(elem).replaceWith(replacementCanvasImgs[i]);
+      });
+    }
+
+    // .addBack('video') handles case when the element itself is a video.
+    var replacementVideoImgs = [];
+    $element.find('video').addBack('video').map(function(i, elem) {
+      var $elem = $(elem);
+      var canvas = htmlTools.cloneDomItem($elem, "<canvas>");
+      canvas[0].getContext('2d').drawImage(elem, 0, 0, $elem.width(), $elem.height());
```

Comment:
*Cool* – I didn't know that the drawImage method could take Dom elements as arguments.  I just looked it up in the [canvas drawImage api](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage)